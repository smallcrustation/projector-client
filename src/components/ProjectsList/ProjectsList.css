.ProjectsList{
  border: 1px solid black;
}

.ProjectsList__header{
  /* text-align: center; */
  font-size: 29px;
  margin: 10px 0 15px 0;
  padding: 0 20px 3px;
  background-image: linear-gradient(to right, black, black);
  /* background-position: bottom center ; */
  background-position: bottom left;
  background-repeat: no-repeat;
  background-size: 35% 2px;
}

.ProjectsList__li{
  font-size: 20px;
  background-color: white;
  padding: 10px 20px;
  margin: 10px 0 0;
  border-left: 2px solid black;
  border-top: 2px solid black;
  border-bottom: 2px solid black;
  transition: 0.3s;
}

.ProjectsList__li:hover{
  background-color: lightgray;
}

.ProjectsList__ul{
  background-color: crimson;
  margin: 0;
  padding: 1px 0 10px 10px;
}

.ProjectsList a:visited,.ProjectsList a:visited{
  color: black;
}