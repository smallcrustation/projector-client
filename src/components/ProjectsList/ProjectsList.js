import React from 'react'
import { Link } from 'react-router-dom'

import './ProjectsList.css'

export default class ProjectsList extends React.Component {
  state = { error: null }

  onClickNewProject = () => {
    console.log('Create Project')
    return this.props.history.push('./CreateProject')
  }

  render() {
    // const { error } = this.state
    return (
      <div className="ProjectsList">
        <h3>Projects</h3>
        <section>
          <ul>
            <li>
              <Link to="/project/1">Project 1</Link>
            </li>
            <li>
              <Link to="/project/2">Project 2</Link>
            </li>
            <li>
              <Link to="/project/3">Project 3</Link>
            </li>
          </ul>
        </section>
      </div>
    )
  }
}
