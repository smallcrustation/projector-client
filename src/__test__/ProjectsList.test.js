import React from 'react'
import ReactDOM from 'react-dom'

import ProjectsList from '../components/ProjectsList/ProjectsList'
import renderer from 'react-test-renderer'

describe('App Testing', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div')

    ReactDOM.render(<ProjectsList />, div)

    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders with no crashing and matches snapshot', () => {
    const tree = renderer.create(<ProjectsList />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
