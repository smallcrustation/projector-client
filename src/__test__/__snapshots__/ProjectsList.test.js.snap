// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Testing Renders with no crashing and matches snapshot 1`] = `
<div
  className="ProjectsList"
>
  <h3
    className="ProjectsList__header"
  >
    Projects
  </h3>
  <section>
    <ul
      className="ProjectsList__ul"
    >
      <p>
        No Consumer Context
      </p>
    </ul>
  </section>
</div>
`;
